digitalocean/datasource_digitalocean_container_registry.go

```diff
@@ -0,0 +1,78 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+const RegistryHostname = "registry.digitalocean.com"
+
+func dataSourceDigitalOceanContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDigitalOceanContainerRegistryRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "name of the container registry",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			//TODO: Need better name?
+			"write": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"server_url": {
```

Comment:
Bit of a nitpick, but since this isn't actually a URL, maybe go with something like `host`?

Reply:
We went with the key that shows up in `docker.json`. We could change it to `host` though. 
```
{
	"ServerURL": "https://index.docker.io/v1",
	"Username": "david",
	"Secret": "passw0rd1"
}
```

Reply:
I'd keep the current name if that's the spec.

Reply:
Interesting... That's a very different format than what is returned by our API: https://developers.digitalocean.com/documentation/v2/#get-docker-credentials-for-container-registry

Reply:
So if you're using a credential store for docker is looks like what I posted https://docs.docker.com/engine/reference/commandline/login/#login-to-a-self-hosted-registry#credentials-store but the regular `.docker/config.json` looks like what our API returns. I'm not married to the attribute name, do maybe `docker_host`?

Reply:
Also Kamal added export for this variable in `godo` https://github.com/digitalocean/godo/pull/306 so I'll be changing it to use this as well.